frontend/src/components/useOutsideClick/index.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from "react";
+
+const useOutsideClick = (ref, handler) => {
+  useEffect(() => {
+    const listener = event => {
+      if (ref.current.contains(event.target)) {
+        return;
+      }
+      handler(event);
+    };
+
+    document.addEventListener("mousedown", listener);
+    document.addEventListener("touchstart", listener);
+
+    return () => {
+      document.removeEventListener("mousedown", listener);
+      document.removeEventListener("touchstart", listener);
+    };
+  }, [ref, handler]);
```

Comment:
Listening for ref changes doesn't really work: https://reactjs.org/docs/hooks-reference.html#useref

```
Keep in mind that useRef doesn’t notify you when its content changes. Mutating the .current property doesn’t cause a re-render. If you want to run some code when React attaches or detaches a ref to a DOM node, you may want to use a callback ref instead.
```

You should try and use e ref callback instead (let me know if you need some help)

Reply:
and yes, i have a callback there.
```const listener = event => {
      if (ref.current.contains(event.target)) {
        return;
      }
      handler(event);
    };```

https://medium.com/@kevinfelisilda/click-outside-element-event-using-react-hooks-2c540814b661
@tudorgergely